packages/core/src/components/OverflowTooltip/OverflowTooltip.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Tooltip } from '@material-ui/core';
+import React, { useEffect, useRef, useState } from 'react';
+
+type Props = {
+  title: string | number | boolean;
+  width?: string;
+  children?: React.ReactNode;
+};
+
+export const OverflowTooltip = (props: Props) => {
+  const { title, width, children } = props;
+
+  const divElementRef = useRef<HTMLDivElement>(null);
+  const [hoverStatus, setHover] = useState(false);
+
+  const compareSize = () => {
+    const textElement = divElementRef.current;
+    if (textElement) {
+      const compare = textElement.scrollWidth > textElement.clientWidth;
+      setHover(compare);
+    }
+  };
+
+  // compare once and add resize listener on "componentDidMount"
+  useEffect(() => {
+    compareSize();
+    window.addEventListener('resize', compareSize);
+  }, []);
+
+  // remove resize listener again on "componentWillUnmount"
+  useEffect(
+    () => () => {
+      window.removeEventListener('resize', compareSize);
```

Comment:
You just return this function from the useEffect above, no need to make two of them. Also, move the compareSize fn inside the useEffect too in my opinion, keeps everything contained in one place and doesn't go through the trouble of recreating that function unnecessarily on every draw. Adding setHover as a dependency of the effect is fine if the linter complains, but I think it won't

Reply:
Agreed specific width is far from ideal, at least it should be calculated dynamically. Need help from css expert.

on the compareSize will try with single useEffect.